src/nextcloud/talk/recording/Participant.py

```diff
@@ -292,8 +301,12 @@ def startFirefox(self, width, height, env):
 
         service = FirefoxService(
             env=env,
+            executable_path=driverPath,
         )
 
+        if binaryLocation is not None:
+            options.binary_location = binaryLocation
```

Comment:
Checking that `binary_location` is not `None` is right, as `binary_location` is expected (in Selenium >= 4.12 for Firefox, 4.11 for Chromium) to be a string and passing `None` fails. However, if the browser is not in $PATH setting `binary_location = ""` for Firefox in Selenium >= 4.12 and < 4.17 would also fail (rather than downloading it). In other versions it should not make any difference to pass an empty string or not setting the option, so it would be safer to pass the argument if it is neither `None` nor an empty string:
```
        if browserPath:
            options.binary_location = browserPath
```